pkg/storage/store.go

```diff
@@ -3087,6 +3088,12 @@ func sendSnapshot(
 	if err != nil {
 		return errors.Wrapf(err, "range=%s: remote failed to apply snapshot", header.RangeDescriptor.RangeID)
 	}
+	// NB: wait for EOF which ensures that all processing on the server side has
+	// completed (such as defers that might be run after the previous message was
+	// received).
+	if _, err := stream.Recv(); err != io.EOF {
```

Comment:
This is rather odd for two reasons 1. the resp is ignored, and 2. I imagine it would be part of a receiver loop